### DISCUSSION TITLE
Quarkus at Devoxx Belgium

### DISCUSSION THREAD

**Discussion by insectengine** [quarkusio/quarkus#28236] [Events]
Devoxx Belgium is a 5-days conference where the developers community get together and explore the latest technology advancements with some of the most inspiring speakers in our sector.

Diverse, local and global talent introduce the newest and most vital content from the development world, with a range of sessions covering Java, Cloud, Big Data, Security, Architecture, Artificial Intelligence, Machine Learning, Robotics, Programming Languages, Methodologies and Developer Culture.

Devoxx Belgium expands your knowledge base, sharpens your skills and provides hands-on experience with the latest technologies.  Next to the obvious networking opportunities with peers and our partners.

Type: In-Person
Date: October 10-14, 2022
https://devoxx.be/search-results/?query=quarkus

### Quarkus Related Sessions and Content

**[A Java developer Journey into Apache Cassandra™](https://devoxx.be/talk/?id=16122)**
**Speaker: Cedrick Lunven**
Description: Apache Cassandra™ is a distributed NoSQL distributed database deployed extensively at web giants like Apple, Netflix or Uber.
During this deep-dive session, we will give you everything you need to master this technology: from architecture to data modeling, from drivers to best practices. Through practical labs requiring no installation, we will browse and run applications implemented with Spring Boot, Quarkus, and Micronaut in order to see how Apache Cassandra™ can be used in modern java applications. Take home 3 working projects at home.

**[Build a modern Java Serverless Quarkus Application](https://devoxx.be/talk/?id=52804)**
**Speakers: Kevin Azijn & Vinicius Senger**
Description: This workshop will guide developers to build a modern, efficient, elegant and sustainable serverless application from scratch to manage personal expenses / banking account using Quarkus, AWS Lambda, CDK and other Cloud services for CI/CD, observability and optional AI/ML integration features.

If you are a Java EE developer looking forward to update your knowledge about modern Java this is the right workshop for you and in you will have a complete reference application backend with the most used Java technologies for cloud native applications and serverless that you can replicate and adapt it to your new Java projects.

**[How to avoid common pitfalls with modern microservices testing](https://devoxx.be/talk/?id=6661)**
**Speakers: Holly Cummins & Eric Deandrea**
**[View the Session Recording](https://youtu.be/vYwkDPrzqV8)**

Description: Have you ever wondered why your microservices break in production after all the tests have passed during CI/CD? Have you ever wondered if you’re testing too much or too little? If you want to learn about common pitfalls and how to catch them at build time rather than at runtime, this session is for you!

In this session we’ll share some common battle scars from our experiences in the field. Additionally, we will introduce the testing pyramid and explain why it is important in helping to continuously deploy microservices independently without fear. We will demonstrate, through a test-driven development approach, how the combination of Pact contract testing combined with Quarkus developer joy can help prevent your microservices from breaking in production.

**[Quarkus Community BOF](https://devoxx.be/talk/?id=1904)**
**Speaker: Dimitris Andreadis**
Description: Quarkus, the Black Swan of Java, is getting more and more popularity for its unbeatable Performance characteristics and the first rate Developer Experience it offers. With more than 630+ contributors to the main project, its rapidly expanding extension ecosystem, and a continuous stream of innovation Quarkus is leading the way in the Kubernetes Native Java Application development space.

Come to this Quarkus Community BOF to discuss the present and future of Quarkus with distinguished members of the Quarkus Development Team.

**[Quarkus, Micronaut and Spring Boot fighting in the Cloud](https://devoxx.be/talk/?id=13115)**
**Speakers: Yoshio Terada, Mark Heckler, & Antonio Goncalves**
Description: On the right corner, Quarkus, goes into his challenge against Micronaut the lightweight champion, and pound-for-pound with Spring Boot the best microservice framework in the world. Who will win the world “Cloud Native Java Runtime” title fight?

In this 3h long workshop you will bring your own laptop, install all the needed tools and be ready to develop from scratch three microservices: one written with Quarkus, one with Spring Boot and one with Micronaut. Add some business logic, access a database in a reactive and imperative manner, run them on HotSpot, compile them with GraalVM, containerize them with Docker and deploy them to the cloud. Then, add some load testing, scale and monitor them. At the end of this workshop you will know which of these microservices is the most appropriate in terms of performance, resource consumption, and startup time… or maybe you won’t ;o)

For this workshop you will need to install:
Git
Java 11, Java 17 and GraalVM
Your own IDE
Docker desktop
An HTTP client (cUrl, wget, Postman)
Have a Docker Hub account
Have a GitHub account
Have an Azure account (we will give you some credits)

**[Serverless Integration with Camel Quarkus](https://devoxx.be/talk/?id=49201)**
**Speaker: Kevin Dubois**
**[View the Session Recording](https://www.youtube.com/watch?v=FBWgbhp8FG8&feature=youtu.be)**
When you build an application you will undoubtedly need to integrate it with other applications and systems. The Apache Camel project’s goal is to simplify the way we do integration, using the book “Enterprise Integration Patterns” as the gold standard.

Apache Camel has been around for a while, but it is more relevant now than it has ever been. In this session we’ll show you how Camel can leverage Quarkus’ fast startup time and developer experience.

As a bonus it can make use of technologies like serverless computing (eg. Knative) and data streaming (eg. Kafka), to remove bottlenecks, integrate faster and better than ever before, and save resources and money on top of it.

Attendees should come away from this event with the following:

Understanding of what Serverless Enterprise Integration means and how to deliver better software faster
How serverless technologies like Knative can optimize resource usage and scale rapidly to respond to fluctuating demand
How the next iteration of Apache Camel (based on Quarkus) can leverage serverless computing and event driven architectures to do integration when and where it’s needed

**[Event-driven autoscaling for Serverless Java](https://devoxx.be/schedule/speaker-details/?id=3902)**
**Speaker: Daniel Oh**
**[View the Session Recording](https://youtu.be/TwO5b7YCbTU)**

Kubernetes makes it possible to autoscale various business use cases from the web apps to mobile, IoT edge streaming, and AI/ML in more reliable and stable ways. One caveat of the Kubernetes autoscaling is based on hardware resource utilization (CPU, memory) through Horizontal Pod Autoscaling. This causes a new challenge to build an event-driven serverless Java on Kubernetes because the event metrics from multiple event sources (e.g., Apache Kafka, AWS SQS) are more relevant than a pod's CPU usage for deciding when applications need to be scaled out and in. 

Fortunately, KEDA and Knative on Kubernetes are designed to solve this challenge by autoscaling both standard apps and serverless by event metrics in a separate way. This session will teach you how to redesign your Kubernetes autoscaling architecture by event-driven metrics from Apache Kafka over standard resources (CPU, Memory) with Knative and KEDA integration for serverless Java using Quarkus.

**Comment by kdubois** [comment]
Does Camel Quarkus count too? In that case there's my session as well :) 


**[Serverless Integration with Camel Quarkus](https://devoxx.be/talk/?id=49201)** 
**Speaker: Kevin Dubois**
When you build an application you will undoubtedly need to integrate it with other applications and systems. The Apache Camel project’s goal is to simplify the way we do integration, using the book “Enterprise Integration Patterns” as the gold standard. 

Apache Camel has been around for a while, but it is more relevant now than it has ever been. In this session we’ll show you how Camel can leverage Quarkus’ fast startup time and developer experience. 

As a bonus it can make use of technologies like serverless computing (eg. Knative) and data streaming (eg. Kafka), to remove bottlenecks, integrate faster and better than ever before, and save resources and money on top of it.

Attendees should come away from this event with the following:

Understanding of what Serverless Enterprise Integration means and how to deliver better software faster
How serverless technologies like Knative can optimize resource usage and scale rapidly to respond to fluctuating demand
How the next iteration of Apache Camel (based on Quarkus) can leverage serverless computing and event driven architectures to do integration when and where it’s needed

**Reply by insectengine** [reply]
Yes indeed, added to the list!

**Comment by edeandrea** [comment]
All the sessions from Devoxx Belgium can be viewed at https://www.youtube.com/playlist?list=PLRsbF2sD7JVolUH45EkGXsT-3spU7cqnS

@insectengine Maybe it would be good to post the links to individual sessions up on the main post of this discussion?

Here's a couple that I found:
- Mine & Holly's: https://youtu.be/vYwkDPrzqV8
- Dan Oh: https://youtu.be/TwO5b7YCbTU
- Kevin Dubois: https://youtu.be/FBWgbhp8FG8
